module5/Publisher.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+from typing import List
+
+from module5.ArabicText import ArabicText
+from module5.InputException import InputException
+from module5.News import News
+from module5.PrivateAd import PrivateAd
+from module5.Publication import Publication
+
+
+class Publisher:
+
+    @staticmethod
+    def get_publication_type() -> str:
+        pub_type = input("Welcome to Publisher!\n"
+                         "Please enter publication one of the follow type: News/Private Ad/Arabic Text: ")
+        if pub_type not in Publication.PUBLICATION_TYPES:
+            raise InputException("Publication type is not supported.")
+        return pub_type
+
+    @staticmethod
+    def get_publication_text() -> List[str]:
+        row_count = int(input("Count of text row: "))
+        if row_count <= 0:
+            raise InputException("Row count should be more than 0.")
+        pub_text = []
+        for row in range(0, row_count):
+            try:
+                line = input(f"Type row {row + 1}: ")
+            except EOFError:
+                break
+            pub_text.append(line)
+        return pub_text
+
+    @staticmethod
+    def get_specific_info(pub_type: str) -> str:
+        if pub_type == "News":
+            return input("Enter City of news: ")
+        elif pub_type == "Private Ad":
+            exp_date = input("Enter expiration date in format %d/%m/%Y : ")
```

Comment:
Maybe give user more human-readable format? like yyyy-mm-dd

Reply:
I think it depends on user and, for example, in my usually life I've never use format yyyy-mm-dd :)
Based on unclear requirements I just took format from home task example.
![image](https://user-images.githubusercontent.com/8178906/154455543-41cab2d6-a298-44a8-8058-71cf88458f49.png)
But I agree that in the input message should be written more readable example.